### ISSUE TITLE
IntelliSense auto completion for known format string values

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#29921]
For objects like DateTime or Int32 there's a known list of supported format strings, such as &quot;s&quot; for DateTime or &quot;N0&quot; for Int32. It'd be really helpful if IntelliSense knew about these and gave auto completion for them.

Outside of the few I use regularly I always have to refer back to the docs to see what the available options are along with the resulting output. What I'd like to see is a list of values such as:
o - 2018-09-10T20:44:10.3523872-04:00
s - 2018-09-10T20:44:10

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/332710/intellisense-auto-completion-for-known-format-stri.html
VSTS ticketId: 681722_
_These are the original issue comments:_
(no comments)
_These are the original issue solutions:_
(no solutions)

**Comment by CyrusNajmabadi** [comment]
@sharwell @jinujoseph The virtualchar/embedded-language work i did could be helpful here.  We could detect format strings and properly provide IDE services in them.

**Comment by CyrusNajmabadi** [comment]
@ivanbasov I don't think this has anything to do wit Watch/Immediate.  This is just about general intellisense for VB/C#.  i.e. when you type ```val.Format(" <-- here``` we should have help in intellisense for what you can legally type in that format string.

**Comment by xt0rted** [comment]
This was my request based on a feature of ReSharper which looks like this:

![image](https://user-images.githubusercontent.com/831974/45639435-fd691200-ba7d-11e8-852a-e46e974728d1.png)

More info under `Formatting Specifier Completion` at https://blog.jetbrains.com/dotnet/2013/07/02/code-completion-improvements-in-resharper-8/

**Comment by ivanbasov** [comment]
Thank you for clarifications, @CyrusNajmabadi and @xt0rted ! Then, it should be implemented in Roslyn.

**Comment by CyrusNajmabadi** [comment]
@olegtk @dpoeschl  Completion has the facility to place gray 'suffix' text after the complation item.  However, i don't know if there's any facility to give the right-aligned example.   Can you guys let me know if the VS completion system actually has a way of providing such a thing.  It would also be *super* useful for the regex completion PR here: https://github.com/dotnet/roslyn/pull/27694

**Comment by olegtk** [comment]
I don't believe we support this. @AmadeusW?
What about simpler form, were "long date" also goes into the suffix?
[icon] D _long date, e.g. Thursday, 10 April 2008_

**Comment by CyrusNajmabadi** [comment]
@olegtk Is there any way for a 3rd party to get cutomization control here?  

Note: as per https://github.com/dotnet/roslyn/pull/27694 i've found the suffix a bit not-great.  Specifically, because it is left aligned, instead of right aligned.  It ends up hurting readability here. 

Would it be difficult to be able to add and expose this through the editor layer somehow?

**Comment by olegtk** [comment]
@AmadeusW ?

**Comment by JeWaVe** [comment]
Please implement that feature for Single and Double as well. 
"G", "F17" etc drive me crazy everytime

**Comment by Dreamescaper** [comment]
I'd like to try to implemented that. 
Is that OK to be picked up? Or still requires some review?
